### PR TITLE
PHPunit speed up attempt

### LINKED ISSUE #191: Improve tests suite performance
Tests should be fast, there not actually. Tests in UI are very slow. A way to make it faster while keeping the use cases covered is needed

### PR DESCRIPTION
**Pull request by jorge07**
Fixes #191
### Initial results

```
docker-compose -f docker-compose.yml -f etc/ci/docker-compose.yml exec -T php sh -lc "./vendor/bin/phpunit --coverage-clover build/logs/clover.xml"
PHPUnit 9.3.0 by Sebastian Bergmann and contributors.

Testing 
...............................................................   63 / 63 (100%)

Time: 08:52.797, Memory: 48.00 MB

OK (63 tests, 130 assertions)

Generating code coverage report in Clover XML format ... done [00:00.050]

Generating code coverage report in HTML format ... done [00:00.374]
```

## Refactoring some code after some blackfire help

```
docker-compose -f docker-compose.yml -f etc/ci/docker-compose.yml exec -T php sh -lc "./vendor/bin/phpunit --coverage-clover build/logs/clover.xml"
PHPUnit 9.3.0 by Sebastian Bergmann and contributors.

Testing 
...............................................................   63 / 63 (100%)

Time: 07:19.836, Memory: 48.00 MB

OK (63 tests, 130 assertions)

Generating code coverage report in Clover XML format ... done [00:00.038]

Generating code coverage report in HTML format ... done [00:00.345]
```

### After replace coverage from xdebug to pcov

```
docker-compose -f docker-compose.yml -f etc/ci/docker-compose.yml exec -T php sh -lc "php -dpcov.enabled='1' -dpcov.directory=. -dpcov.exclude='~vendor~' ./vendor/bin/phpunit --coverage-clover build/logs/clover.xml"
PHPUnit 9.3.0 by Sebastian Bergmann and contributors.

Testing 
...............................................................   63 / 63 (100%)

Time: 00:04.752, Memory: 52.00 MB

OK (63 tests, 130 assertions)

Generating code coverage report in Clover XML format ... done [00:00.022]

Generating code coverage report in HTML format ... done [00:00.231]
```

### CODE REVIEW

**Comment by jorge07** [comment]
I'll add one suite without coverage

File: src/Domain/User/ValueObject/Auth/HashedPassword.php [hunk 1]
```diff
@@ -45,6 +49,10 @@ private static function hash(string $plainPassword): string
     {
         Assertion::minLength($plainPassword, 6, 'Min 6 characters password');
 
+        if (self::isTestSuite()) {
+            return $plainPassword;
+        }
+
```
Comment (Lutacon) [inline src/Domain/User/ValueObject/Auth/HashedPassword.php:55]: Why is this needed?

**Review by Lutacon — APPROVED** [review]
LGTM.

What a change from xdebug to pcov 👏🏼

File: src/Domain/User/ValueObject/Auth/HashedPassword.php (hunk 1 above)
Comment (jorge07) [reply src/Domain/User/ValueObject/Auth/HashedPassword.php:55]: https://blackfire.io/profiles/40247026-d471-4234-902c-e56990cea0f9/graph?settings%5Bdimension%5D=wt&settings%5Bdisplay%5D=landscape&settings%5BtabPane%5D=nodes&selected=&callname=main()

bcrypt cost in test suite is 1, too high 2, unnecessary

Comment (Lutacon) [reply src/Domain/User/ValueObject/Auth/HashedPassword.php:55]: Ok 👍🏼 . ( Link doesn't work 😞 )

Comment (jorge07) [reply src/Domain/User/ValueObject/Auth/HashedPassword.php:55]: Screenshot 2020-09-21 at 13 19 43

Comment (jorge07) [reply src/Domain/User/ValueObject/Auth/HashedPassword.php:55]: 10+5s just encrypting and decrypting
